### PR TITLE
Add `list-pr-for-branch` feature

### LINKED ISSUE #3713: Add a link from the branch commits page to the PR page
Branch commits page: `https://github.com/<user>/<repo>/commits/<branch-name>`

Would be nice to add a link from this page to the PR page, if a PR exists for the branch.

### PR DESCRIPTION
**Pull request by yakov116**
Closes #3713

## Test URLs
https://github.com/yakov116/TestR/commits/yakov116-patch-2
https://github.com/yakov116/TestR/commits/yakov116-patch-3
https://github.com/sindresorhus/refined-github/commits/list-pr-for-branch

## Screenshot
image

### CODE REVIEW

**Comment by fregante** [comment]
The test URL is incorrect by the way, it should only be `/commits/` I think

**Comment by yakov116** [comment]
> The test URL is incorrect by the way, it should only be `/commits/` I think

Updated

File: readme.md [hunk 1]
```diff
@@ -170,6 +170,7 @@ Thanks for contributing! 🦋🙌
 - [](# "refined-github.css") [Reduces tabs’ size to 4 spaces instead of 8.](https://cloud.githubusercontent.com/assets/170270/14170088/d3be931e-f755-11e5-8edf-c5f864336382.png)
 - [](# "esc-to-deselect-line") [Adds a keyboard shortcut to deselect the current line: <kbd>esc</kbd>.](https://github.com/sindresorhus/refined-github/issues/1590)
 - [](# "vertical-front-matter") [Shows Markdown front matter as vertical table.](https://user-images.githubusercontent.com/44045911/87251695-26069b00-c4a0-11ea-9077-53ce366490ed.png)
+- [](# "list-pr-for-branch") [On branch commit lists, shows the PR that touch the current branch.](https://user-images.githubusercontent.com/16872793/119760295-b8751a80-be77-11eb-87da-91d0c403bb49.png)
```
Comment (fregante) [inline readme.md:173]:
```suggestion
- [](# "list-pr-for-branch") [On branch commit lists, shows the PR that touches the current branch.](https://user-images.githubusercontent.com/16872793/119760295-b8751a80-be77-11eb-87da-91d0c403bb49.png)
```

File: source/features/list-pr-for-branch.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import * as pageDetect from 'github-url-detection';
+import {GitMergeIcon, GitPullRequestIcon} from '@primer/octicons-react';
+
+import features from '.';
+import getDefaultBranch from '../github-helpers/get-default-branch';
+import {addAfterBranchSelector} from './latest-tag-button';
+import {getPullRequestsAssociatedWithBranch} from './show-associated-branch-prs-on-fork';
+
+// Taken from https://github.com/fregante/github-issue-link-status/blob/98792f2837352bacbf80664f3edbcec8e579ed17/source/github-issue-link-status.js#L10
+const stateColorMap = {
```
Comment (fregante) [inline source/features/list-pr-for-branch.tsx:12]: Does this exist in another feature? It can be exported from there

File: readme.md (hunk 1 above)
Comment (fregante) [reply readme.md:173]: The PR is untested, but looks good otherwise

File: source/features/list-pr-for-branch.tsx (hunk 2 above)
Comment (yakov116) [reply source/features/list-pr-for-branch.tsx:12]: No. Those are different class names

**Pull request merged**